File: gx-sneaker-ui/src/services/sanPhamService.js
import axios from "axios"

const API = "http://localhost:8080/api/san-pham"

export const getAll = () => axios.get(API)

export const getById = (id) =>
  axios.get(`${API}/${id}`)

export const create = (data) =>
  axios.post(API, data)

export const update = (id, data) =>
  axios.put(`${API}/${id}`, data)

export const remove = (id) =>
  axios.delete(`${API}/${id}`)
export const getFeaturedProducts = () =>
  axios.get(`${API}/featured`)

export const getNewProducts = () =>
  axios.get(`${API}/new`)

File: src/main/java/com/gxsneaker/gxsneaker/service/SanPhamService.java
package com.gxsneaker.gxsneaker.service;

import com.gxsneaker.gxsneaker.dto.SanPhamDTO;

import java.util.List;

public interface SanPhamService {

    List<SanPhamDTO> getAll();

    SanPhamDTO getById(Long id);

    SanPhamDTO create(SanPhamDTO dto);

    SanPhamDTO update(Long id, SanPhamDTO dto);

    void delete(Long id);
    List<SanPhamDTO> search(String keyword);
    List<SanPhamDTO> getNewProducts();

    List<SanPhamDTO> getFeaturedProducts();
    List<SanPhamDTO> getNewestProducts();
    void stopSelling(Long id);
    List<SanPhamDTO> getAllActive();
}